py/desispec/trace_shifts.py

```diff
@@ -117,29 +117,33 @@ def write_traces_in_psf(input_psf_filename,output_psf_filename,xcoef,ycoef,wavem
     psftype=psf_fits[0].header["PSFTYPE"]
     if psftype=="GAUSS-HERMITE" :             
         
-        i=np.where(psf_fits[1].data["PARAM"]=="X")[0][0]
-        ishape=psf_fits[1].data["COEFF"][i].shape
-        if ishape != xcoef.shape :
-            log.warning("xcoef from file and from arg don't have same shape : %s != %s"%(str(ishape),str(xcoef.shape)))
-        
-        n0=min(ishape[0],xcoef.shape[0])
-        n1=min(ishape[1],xcoef.shape[1])
-        psf_fits[1].data["COEFF"][i] *= 0.
-        psf_fits[1].data["COEFF"][i][:n0,:n1]=xcoef[:n0,:n1]
-        psf_fits[1].data["WAVEMIN"][i]=wavemin
-        psf_fits[1].data["WAVEMAX"][i]=wavemax
+        i=np.where(psf_fits["PSF"].data["PARAM"]=="X")[0]
+        if i.size>0 :
+            i=i[0]
+            ishape=psf_fits["PSF"].data["COEFF"][i].shape
+            if ishape != xcoef.shape :
+                log.warning("xcoef from file and from arg don't have same shape : %s != %s"%(str(ishape),str(xcoef.shape)))
+
+            n0=min(ishape[0],xcoef.shape[0])
+            n1=min(ishape[1],xcoef.shape[1])
+            psf_fits["PSF"].data["COEFF"][i] *= 0.
+            psf_fits["PSF"].data["COEFF"][i][:n0,:n1]=xcoef[:n0,:n1]
+            psf_fits["PSF"].data["WAVEMIN"][i]=wavemin
+            psf_fits["PSF"].data["WAVEMAX"][i]=wavemax
         
         
-        i=np.where(psf_fits[1].data["PARAM"]=="Y")[0][0]
-        ishape=psf_fits[1].data["COEFF"][i].shape
-        if ishape != ycoef.shape :
-            log.warning("xcoef from file and from arg don't have same shape : %s != %s"%(str(ishape),str(ycoef.shape)))
-        n0=min(psf_fits[1].data["COEFF"][i].shape[0],ycoef.shape[0])
-        n1=min(psf_fits[1].data["COEFF"][i].shape[1],ycoef.shape[1])
-        psf_fits[1].data["COEFF"][i] *= 0.
-        psf_fits[1].data["COEFF"][i][:n0,:n1]=ycoef[:n0,:n1]
-        psf_fits[1].data["WAVEMIN"][i]=wavemin
-        psf_fits[1].data["WAVEMAX"][i]=wavemax
+        i=np.where(psf_fits["PSF"].data["PARAM"]=="Y")[0]
+        if i.size>0 :
```

Comment:
This code is fine, but for future reference, `if "Y" in psf_fits["PSF"].data["PARAM"]:` or `if np.count_nonzero(psf_fits["PSF"].data["PARAM"]=="Y") > 0:` might have been clearer.